gap/semigroups/semieunit.gi

```diff
@@ -0,0 +1,715 @@
+#############################################################################
+##
+#W  semieunit.gi
+#Y  Copyright (C) 2017                                    Christopher Russell
+##
+##  Licensing information can be found in the README file of this package.
+##
+#############################################################################
+#
+# This code is based on the description of McAlister triple semigroups found
+# in Fundamentals of Semigroup Theory by John Howie.
+#
+#############################################################################
+# Methods for creating McAlister triple semigroups
+#############################################################################
+InstallMethod(McAlisterTripleSemigroup,
+"for a perm group, action, digraph, and digraph",
+[IsGroup, IsFunction, IsDigraph, IsDigraph],
```

Comment:
Sorry, I think I forgot to change the label at some point. It should work for any finite group, when there is also a valid action given as an argument.